desc/objectives/_neoclassical.py

```diff
@@ -256,7 +250,242 @@ def compute(self, params, constants=None):
+        Whether to vectorize part of the computation. Default is true.
+    Y_B : int
+        Desired resolution for algorithm to compute bounce points.
+        Default is double ``Y``. Something like 100 is usually sufficient.
+        Currently, this is the number of knots per toroidal transit over
+        to approximate |B| with cubic splines.
+    num_transit : int
+        Number of toroidal transits to follow field line.
+        For axisymmetric devices, one poloidal transit is sufficient. Otherwise,
+        assuming the surface is not near rational, more transits will
+        approximate surface averages better, with diminishing returns.
+    num_well : int
+        Maximum number of wells to detect for each pitch and field line.
+        Giving ``None`` will detect all wells but due to current limitations in
+        JAX this will have worse performance.
+        Specifying a number that tightly upper bounds the number of wells will
+        increase performance. In general, an upper bound on the number of wells
+        per toroidal transit is ``Aι+B`` where ``A``,``B`` are the poloidal and
+        toroidal Fourier resolution of |B|, respectively, in straight-field line
+        PEST coordinates, and ι is the rotational transform normalized by 2π.
+        A tighter upper bound than ``num_well=(Aι+B)*num_transit`` is preferable.
+        The ``check_points`` or ``plot`` methods in ``desc.integrals.Bounce2D``
+        are useful to select a reasonable value.
+    num_quad : int
+        Resolution for quadrature of bounce integrals. Default is 32.
+    num_pitch : int
+        Resolution for quadrature over velocity coordinate. Default is 64.
+    Nemov : bool
+        Whether to use the Γ_c as defined by Nemov et al. or Velasco et al.
+        Default is Nemov. Set to ``False`` to use Velascos's.
+
+        Nemov's Γ_c converges to a finite nonzero value in the infinity limit
+        of the number of toroidal transits. Velasco's expression has a secular
+        term that drives the result to zero as the number of toroidal transits
+        increases if the secular term is not averaged out from the singular
+        integrals. Currently, an optimization using Velasco's metric may need
+        to be evaluated by measuring decrease in Γ_c at a fixed number of toroidal
+        transits.
+
+    """
+
+    __doc__ = __doc__.rstrip() + collect_docs(
+        target_default="``target=0``.",
+        bounds_default="``target=0``.",
+        normalize_detail=" Note: Has no effect for this objective.",
+        normalize_target_detail=" Note: Has no effect for this objective.",
+        overwrite=_bounce_overwrite,
+    )
+
+    _coordinates = "r"
+    _units = "~"
+    _print_value_fmt = "Γ_c: "
+
+    def __init__(
+        self,
+        eq,
+        target=None,
+        bounds=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        loss_function=None,
+        deriv_mode="auto",
+        *,
```

Comment:
one other thing: we had talked about eventually making all args keyword only for objectives so that we can add new ones and re-order as needed (and so we can make the ordering in the docs match the signature). We could start doing that here, but let's discuss at the meeting tomorrow how we want to proceed with that

Reply:
I second this

Reply:
So from the dev meeting I think we should start doing this going forward. `eq` can be positional but everything else keyword only. We can start doing it for all new objectives, eventually we'll also do it for existing objectives.

Reply:
ok i made issue